plugins/bcollection/_base.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright (c) 2019 Fernando Póvoa (sbrubes)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+import logging
+import os.path
+
+import xl.settings
+
+from _utils import (
+    Event,
+    PluginInfo,
+    get_database_path_from_beets_config,
+    get_default_view_pattern,
+)
+
+#: str: Base dir to plugin
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+#: PluginInfo: Plugin info
+PLUGIN_INFO = PluginInfo(BASE_DIR)
+
+
+def get_name(name, join_exp='-'):
+    """
+        Gets a plugin "unique" attribute name
+        :param name: str - attribute name
+        :param join_exp: str ('-' as default)
+        :return: str
+    """
+    return join_exp.join(['plugin', PLUGIN_INFO.id, name])
+
+
+def create_event(name):
+    """
+        Creates a Event
+        :param name: str
+        :return: Event
+    """
+    return Event(get_name(name))
+
+
+def get_path(*paths):
+    """
+            Get a path on plugin
+            :param paths:
+            :return: str (joining paths with base dir)
+        """
+    return os.path.join(BASE_DIR, *paths)
+
+
+class Logger:
```

Comment:
Why are you writing your own logger class? You could just write

    LOGGER = logging.getLogger(__name__)
    […]
    LOGGER.info('some info')
    LOGGER.error('this code failed')

This is how we're doing it all over our codebase.